dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -1108,7 +1109,25 @@ private void processWesCommands(final List<String> args) {
     }
 
     /**
-     * Launches tools and workflows.
+     * Prints a warning if Docker isn't running. Docker is not always needed. If a workflow or tool uses Docker and
+     * it is not running, it fails with a cryptic error. This should make the problem more obvious.
+     */
+    void checkIfDockerRunning() {
```

Comment:
Ah, there's a client for this kind of thing. 
See https://github.com/spotify/docker-client/blob/master/docs/user_manual.md#display-system-wide-information
and example in our code https://github.com/dockstore/dockstore/blob/54ce647c96f2b96be555d114fadce5ad102d6b54/dockstore-event-consumer/src/main/java/io/dockstore/consumer/handler/DOIHandler.java#L100

Reply:
Oh okay, that does simplify things

Reply:
I wonder if this is worth it because of the dependencies it brings in. I tried builds without and with this change, and the client JAR went up from ~78.05M to ~85.86M. That doesn't matter for the webservice, but I think the CLI should be as slim as possible.

Or is the client doing some additional checks that can't be captured just running the docker command-line?

Should we go back to just using ProcessBuilder, I would recommend using `docker info` instead of `docker ps`, even though I had suggested `docker ps` -- I think `info` will be lighter weight.

Reply:
Hmmm, i can see where you're coming from. But I suspect due to our project's focus on Docker, a Docker client may prove particularly useful. 

Let's approach it this way, no developer has really taken a close look at our jar size. 
@coverbeck please write up a ticket for a developer to really take a look at it, I suspect that there is low hanging fruit like exploring the use of minimizejar https://maven.apache.org/plugins/maven-shade-plugin/examples/includes-excludes.html  and doing something like a treemap to identify truly large and un-necessary dependencies.

Reply:
#2589